### PR TITLE
Make OCP deployer CI compatible

### PR DESCRIPTION
**Pull request by pebrc**
Fixes https://github.com/elastic/cloud-on-k8s/issues/4420
Fixes https://github.com/elastic/cloud-on-k8s/issues/4319

The first iteration of this revised OCP driver was not really CI compatible: 
* We cannot use `eck-dev` images in CI => change to `eck-ci`
* We need to login into to docker because we cannot rely on a Docker login happening outside of the deployer invocation
* We need to use a shared volume between CI container and client container, which we don't need in local mode because the Make process is not running in a container as well. 
* We need to specify the  OCP version in the Jenkins job we want to test. To conserve resources if the version is not specified it just picks one of the supported versions which are tested round robin during the course of a week

I have tested this on a local Jenkins instance. But I have not yet tested the flow where this job is called from an upstream job as a dependent job.

### CODE REVIEW

File: hack/deployer/runner/ocp.go [hunk 1]
```diff
@@ -425,20 +446,19 @@ func (d *OcpDriver) bucketParams() map[string]interface{} {
...
+		"ClusterStateDirBase": filepath.Base(d.runtimeState.ClusterStateDir),
 		"HomeVolume":          SharedVolumeName(),
 		"GCloudCredsPath":     filepath.Join("/home", GCPDir, ServiceAccountFilename),
 		"OCPToolsDockerImage": d.runtimeState.ClientImage,
 		"Action":              action,
 	}
 	cmd := NewCommand(`docker run --rm \
-		-v {{.HomeVolume	}}:/home \
-		-v {{.ClusterStateDir}}:/config \
+		-v {{.HomeVolume}}:/home \
```
Comment (barkbay) [inline hack/deployer/runner/ocp.go:456]: I think this volume mount can be confusing as it is not the `/home` directory which is mounted here but the home directory of the current user.

**Review by barkbay — APPROVED** [review]
LGTM

File: hack/deployer/runner/ocp.go (hunk 1 above)
Comment (pebrc) [reply hack/deployer/runner/ocp.go]: I tried to make it clearer by adding a comment and renaming the template parameter.

File: hack/deployer/runner/ocp.go [hunk 2]
```diff
@@ -425,20 +446,19 @@ func (d *OcpDriver) bucketParams() map[string]interface{} {
...
 		-v /tmp:/tmp \
 		-e GOOGLE_APPLICATION_CREDENTIALS={{.GCloudCredsPath}} \
-		-e USER_HOME=/home \
+		-e HOME=/home \
```
[the remark this answers is not in the document]
Comment (pebrc) [reply hack/deployer/runner/ocp.go:463]: Mostly so that the GCloud tools pick up the credentials.

**Pull request merged**